### PR TITLE
Fix gatekeeper config update bug

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
When editing the Gatekeeper CR, the config CR doesn't get updated at all. 
Ref: https://issues.redhat.com/browse/ACM-12323
Signed-off-by: yiraeChristineKim <yikim@redhat.com>

### CODE REVIEW

File: controllers/config_helper.go [hunk 1]
```diff
@@ -85,6 +87,77 @@ func createDefaultConfig(ctx context.Context, c client.Client, namespace string,
 	return nil
 }
 
+// Reset Config.Spec.Match and append the default exempt namespaces and
```
Comment (yiraeChristineKim) [inline controllers/config_helper.go:90]: Copy & paste

File: controllers/config_helper.go [hunk 2]
```diff
@@ -95,7 +168,8 @@ func (r *GatekeeperReconciler) initConfig(
 		Resource: "configs",
 	}
 
-	_, err := r.DynamicClient.Resource(configGVR).Get(ctx, "config", metav1.GetOptions{})
+	config, err := r.DynamicClient.Resource(configGVR).
+		Namespace(r.Namespace).Get(ctx, "config", metav1.GetOptions{})
```
Comment (yiraeChristineKim) [inline controllers/config_helper.go:172]: Before, namespace wasn't specified so it caused error loop

File: controllers/config_helper.go [hunk 3]
```diff
@@ -111,6 +185,21 @@ func (r *GatekeeperReconciler) initConfig(
...
 
+	var updatedConfig *v1alpha1.Config
+
+	err = runtime.DefaultUnstructuredConverter.
+		FromUnstructured(config.Object, &updatedConfig)
+	if err != nil {
+		return err
+	}
+
+	err = setExemptNamespaces(ctx, updatedConfig, gatekeeper, r.Log, r.Client, r.Scheme)
```
Comment (yiraeChristineKim) [inline controllers/config_helper.go:196]: Update config when gatekeeper CR updated

**Review by mprahl — APPROVED** [review]
/hold in case @dhaiducek wants to review

**Pull request merged**